src/components/dace_interface.ts

```diff
@@ -107,46 +129,47 @@ export class DaCeInterface {
         if (overridePath && overridePath !== '')
             return this.cleanCmd([overridePath], spaceSafe);
 
+        const fallbackCommand = 'python';
+
         try {
             let pyExt = vscode.extensions.getExtension('ms-python.python');
             if (!pyExt) {
-                // TODO: do we want to tell the user that using the python
-                // plugin might be advisable here?
-                return 'python';
+                vscode.window.showWarningMessage(
+                    `Could not find the Python extension to run the ` +
+                    `DaCe backend, falling back to the regular 'python' ` +
+                    `command. If you have DaCe installed in a virtual ` +
+                    `environment, installing the Python extension is highly ` +
+                    `recommended.`
+                );
+                return fallbackCommand;
             }
 
-            if (pyExt.packageJSON?.featureFlags?.usingNewInterpreterStorage) {
-                if (!pyExt.isActive)
-                    await pyExt.activate();
-                const pyCmd = pyExt.exports.settings.getExecutionDetails ?
-                    pyExt.exports.settings.getExecutionDetails(
-                        uri
-                    ).execCommand :
-                    pyExt.exports.settings.getExecutionCommand(uri);
-                // Ensure spaces in the python command don't trip up the
-                // terminal.
-                if (pyCmd)
-                    return this.cleanCmd(pyCmd, spaceSafe);
-                else
-                    return 'python';
-            } else {
-                let path = undefined;
-                if (uri)
-                    path = vscode.workspace.getConfiguration(
-                        'python',
-                        uri
-                    ).get<string>('pythonPath');
-                else
-                    path = vscode.workspace.getConfiguration(
-                        'python'
-                    ).get<string>('pythonPath');
-                if (!path)
-                    return 'python';
-            }
-        } catch (ignored) {
-            return 'python';
+            if (!pyExt.isActive)
+                await pyExt.activate();
+
+            const environmentsAPI = pyExt.exports.environments;
```

Comment:
nice!

Reply:
Remains to be seen if it is actually more stable, but on my machines it works significantly better. Conda is super annoying here (rant below, feel free to ignore)

<details>
<summary>Rant</summary>
Essentially when you launch a new terminal, the python extension will activate the selected environment in it (so using `conda activate`, if you're using conda). However, you cannot wait for that to finish. So if you programmatically open a terminal and send it a command, if environment activation takes more than a few milliseconds, your command is run before the environment is actually active. That's why `conda run` was needed. The reason it may still be unstable, is that _if_ activation _does_ actually complete fast (within a few ms), then `conda run` is used _after_ environment activation. In that case it does not work properly and will give you import errors. So let's hope for the best.
</details>